onnxruntime/test/providers/provider_test_utils.cc

```diff
@@ -707,6 +714,23 @@ void OpTester::AddInitializers(onnxruntime::Graph& graph) {
   }
 }
 
+#ifdef USE_XNNPACK
+static void AddXNNPackNode(onnxruntime::Graph& graph, Logger& logger) {
```

Comment:
Not quite following this. Why do we need to do this here instead of as part of the usual graph optimization in InferenceSession::Initialize?

Reply:
By default OpTest disables all  graph optimizations. It make sense. If the graph is changed, it might not be what you want to test. However, currently xnnpack relies on graph optimizations. Without this code, I can't re-use the current op tests.